logging/src/commonMain/kotlin/Log.kt

```diff
@@ -1,16 +1,23 @@
 package com.juul.tuulbox.logging
 
-import kotlin.jvm.Volatile
+import co.touchlab.stately.isolate.IsolateState
+
+private class Reference<T : Any>(var value: T)
 
 /** Global logging object. To receive logs, call [dispatcher].[install][DispatchLogger.install]. */
 public object Log {
 
     /** Global log dispatcher. */
     public val dispatcher: DispatchLogger = DispatchLogger()
 
+    private val isolatedTagGenerator = IsolateState { Reference(defaultTagGenerator) }
+
     /** Global tag generator for log calls without explicit tag. */
-    @Volatile
-    public var tagGenerator: TagGenerator = defaultTagGenerator
+    public var tagGenerator: TagGenerator
+        get() = isolatedTagGenerator.access { it.value }
+        set(value) {
+            isolatedTagGenerator.access { it.value = value }
+        }
```

Comment:
If it's OK for the tag generator to be frozen, then `atomic` would be faster (I believe).